Add instructions how to configure Bitbucket Server oAuth2 (#2554)
codenvy/che-docs: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc

:_content-type: PROCEDURE

> :description: Setting up an oauth-2 application link on the Bitbucket Server

Reviewer: ```suggestion
:description: Setting up an OAuth 2.0 application link on the Bitbucket Server
```

:keywords: bitbucket, bitbucket-server, application-link, oauth2

> :navtitle: Setting up an oauth-2 application link on the Bitbucket Server

Reviewer: ```suggestion
:navtitle: Setting up an OAuth 2.0 application link on the Bitbucket Server
```

// :page-aliases:

[id="setting-up-an-oauth-2-application-link-on-the-bitbucket-server"]

> = Setting up an oauth-2 application link on the Bitbucket Server

Reviewer: ```suggestion
= Setting up an OAuth 2.0 application link on the Bitbucket Server
```

> Set up an application link for OAuth 2.0 on the Bitbucket Server.

Reviewer: ```suggestion
Set up an OAuth 2.0 application link on the Bitbucket Server.
```

.Prerequisites

* You are logged in to the Bitbucket Server.
* link:https://www.gnu.org/software/coreutils/base64[`base64`] is installed in the operating system you are using.

.Procedure

. Go to *Administration > Applications > Application links*.
. Select *Create link*.

> . Select *External application*, and then choose *Incoming* as the direction.

Reviewer: Nitpick:
```suggestion
. Select *External application* and *Incoming*.
```

. Enter `pass:c,a,q[{prod-url}]/api/oauth/callback` to the *Redirect URL* field.

> . Chose `Admin - Write`  checkbox in the `Application permissions` section.

Reviewer: ```suggestion
. Select the *Admin - Write*  checkbox in *Application permissions*.
```

> . Click the *Save* button.

Reviewer: ```suggestion
. Click *Save*.
```

> . Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:

Reviewer: ```suggestion
. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```

+
[subs="+quotes,+attributes,+macros"]
----

> $ echo -n '__<github_oauth_client_id>__' | base64

Reviewer: ```suggestion
$ echo -n '__<Bitbucket_Client_ID>__' | base64
```

----

> . Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link Secret:

Reviewer: ```suggestion
. Copy the *Client secret* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```

> . Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link App Secret:

Reviewer: ```suggestion
. Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```

> . Copy the *Client Secret* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:

Reviewer: ```suggestion
. Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link App Secret:
```

+
[subs="+quotes,+attributes,+macros"]
----

> $ echo -n '__<github_oauth_client_secret>__' | base64

Reviewer: ```suggestion
$ echo -n '__<Bitbucket_Client_secret>__' | base64
```
